Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -146,6 +146,42 @@ public enum VersionSetSpecifier: Hashable, CustomStringConvertible {
     }
 }
 
+/// A requirement that a package must satisfy.
+public enum PackageRequirement: Hashable {
+	
+	/// The requirement is specified by the version set.
+	case versionSet(VersionSetSpecifier)
+	
+	/// The requirement is specified by the revision.
+	///
+	/// The revision string (identifier) should be valid and present in the
+	/// container. Only one revision requirement per container is possible
+	/// i.e. two revision requirements for same container will lead to
+	/// unsatisfiable resolution. The revision requirement can either come
+	/// from initial set of constraints or from dependencies of a revision
+	/// requirement.
+	case revision(String)
+	
+	/// Un-versioned requirement i.e. a version should not resolved.
+	case unversioned
+	
+	/// Returns if this requirement pins to an exact version, e.g. a specific
+	/// version or a revision.
+	public var isExact: Bool {
+		switch self {
+		case .versionSet(let vs):
+			if case .exact = vs {
+				return true
+			}
+			return false
+		case .revision(_):
```

Comment:
nit: This can just be `case .revision:` (I know you're just moving the code but would be nice to address this since you're here)